modules/health_quest/app/controllers/health_quest/v0/appointments_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module HealthQuest
+  module V0
+    class AppointmentsController < HealthQuest::V0::BaseController
+      before_action :validate_params, only: :index
+
+      def index
+        res = each_serializer.new(appointments[:data], meta: appointments[:meta])
+        render json: res
+      end
+
+      private
+
+      def appointment_by_id
+        appointment_service.get_appointment_by_id(params[:id])
+      end
+
+      def appointment_service
+        HealthQuest::AppointmentService.new(current_user)
+      end
+
+      def appointments
+        @appointments ||=
+          appointment_service.get_appointments(type, start_date, end_date, pagination_params)
+      end
+
+      def each_serializer
+        "HealthQuest::V0::#{params[:type].upcase}AppointmentsSerializer".constantize
+      end
+
+      def validate_params
+        raise Common::Exceptions::ParameterMissing, 'type' if type.blank?
+        raise Common::Exceptions::InvalidFieldValue.new('type', type) unless %w[va cc].include?(type)
```

Comment:
I think to be consistent with the other helpers, the InvalidFieldValue error should be in the `type` method and the line above should be `raise Common::Exceptions::ParameterMissing, 'type' if params[:type].blank?`
Then maybe this method would be named `validate_params_present`?

Reply:
ok, got it ..

Reply:
Actually, I realized that there is no longer a CC type so since there is only one type, the type() method and some other things are not needed so I got rid of that method and simplified some things